src/PanoViewer/PanoViewer.js

```diff
@@ -801,26 +802,71 @@ export default class PanoViewer extends Component {
 	 * Register the callback on the raf to call _renderLoop every frame.
 	 */
 	_startRender() {
-		this._renderLoop = this._renderLoop.bind(this);
+		if (IS_SAFARI_ON_DESKTOP) {
+			this._renderLoop = this._renderLoopForNextTick.bind(this);
+		} else {
+			this._renderLoop = this._renderLoop.bind(this);
+		}
+
 		this._rafId = window.requestAnimationFrame(this._renderLoop);
 	}
 
-	_renderLoop() {
+	_render() {
 		if (this._photoSphereRenderer) {
 			if (this._quaternion) {
 				this._photoSphereRenderer.renderWithQuaternion(this._quaternion, this._fov);
 			} else {
 				this._photoSphereRenderer.render(this._yaw, this._pitch, this._fov);
 			}
 		}
+	}
+
+	_renderLoop() {
+		this._render();
+
 		this._rafId = window.requestAnimationFrame(this._renderLoop);
 	}
 
+	/**
+	 * MacOS X Safari Bug Fix
+	 * This code guarantees that rendering should be occurred.
+	 *
+	 * In MacOS X(10.14.2), Safari (12.0.2)
+	 * The requestAnimationFrame(RAF) callback is called just after previous RAF callback without term
+	 * only if requestAnimationFrame is called for next frame while updating frame is delayed (~over 2ms)
+	 * So browser cannot render the frame and may be freezing.
+	 */
+	_renderLoopForNextTick() {
+		const before = performance.now();
+
+		this._render();
+
+		const diff = performance.now() - before;
+
+		if (this._rafTimer) {
```

Comment:
@jongmoon 
I think the _renderLoop > _render process happens sequentially. Is there any clearTimeout?